pkg/webhook/server.go

```diff
@@ -82,72 +75,47 @@ type Webhook interface {
 	Validate() error
 }
 
-// NewServer creates a new admission webhook server.
-func NewServer(name string, options ServerOptions) (*Server, error) {
-	as := &Server{
-		Name:          name,
-		sMux:          http.NewServeMux(),
-		registry:      map[string]Webhook{},
-		ServerOptions: options,
-	}
-
-	return as, nil
-}
-
-// setDefault does defaulting for the Server.
-func (s *Server) setDefault() {
-	if len(s.Name) == 0 {
-		s.Name = "default-k8s-webhook-server"
-	}
-	if s.registry == nil {
-		s.registry = map[string]Webhook{}
-	}
-	if s.sMux == nil {
-		s.sMux = http.DefaultServeMux
-	}
+// setDefaults does defaulting for the Server.
+func (s *Server) setDefaults() {
+	s.webhooks = map[string]Webhook{}
+	s.webhookMux = http.NewServeMux()
 	if s.Port <= 0 {
 		s.Port = 443
 	}
 	if len(s.CertDir) == 0 {
-		s.CertDir = path.Join("k8s-webhook-server", "cert")
+		s.CertDir = path.Join("/tmp", "k8s-webhook-server", "serving-certs")
 	}
 }
 
 // Register validates and registers webhook(s) in the server
 func (s *Server) Register(webhooks ...Webhook) error {
+	// TODO(directxman12): is it really worth the ergonomics hit to make this a catchable error?
+	// In most cases you'll probably just bail immediately anyway.
 	for i, webhook := range webhooks {
 		// validate the webhook before registering it.
 		err := webhook.Validate()
 		if err != nil {
 			return err
 		}
-		_, found := s.registry[webhook.GetPath()]
+		_, found := s.webhooks[webhook.GetPath()]
 		if found {
 			return fmt.Errorf("can't register duplicate path: %v", webhook.GetPath())
 		}
-		s.registry[webhook.GetPath()] = webhooks[i]
-		s.sMux.Handle(webhook.GetPath(), webhook)
+		s.webhooks[webhook.GetPath()] = webhooks[i]
+		s.webhookMux.Handle(webhook.GetPath(), webhook)
 	}
 
 	return nil
 }
 
-// Handle registers a http.Handler for the given pattern.
-func (s *Server) Handle(pattern string, handler http.Handler) {
```

Comment:
The original intend was to give the user flexibility to add additional handlers. e.g. `statusz`, `healthz` etc.

Reply:
right, but they shouldn't be doing it on the webhook server.  They should do it elsewhere, or we should let people pass in their own mux.